### PR TITLE
Rename to cborg

### PR DESCRIPTION
**Pull request by bgamari**
This renames the library from `binary-serialise-cbor` to `cborg`, as suggested in #26. In addition, we provide a thin shim re-exporting the interface with the module naming used previously. I expect we will only maintain this shim for a few versions before deprecating it.

### CODE REVIEW

File: src/Serialise/Cborg/FlatTerm.hs-boot [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module Serialise.Cborg.FlatTerm where
+
+import {-# SOURCE #-} Serialise.Cborg.Encoding
+
+type FlatTerm = [TermToken]
+
+data TermToken
+instance Show TermToken
+
+toFlatTerm :: Encoding -> FlatTerm
```
Comment (axman6) [inline src/Serialise/Cborg/FlatTerm.hs-boot]: Not sure if anyone cares, but this (and a few other files) don't end with newlines

**Comment by axman6** [comment]
It looks like the tests are timing out on GHC 7.8, this can be fixed by using the travis_wait command (which just prints `.`'s to stop long running tasks for being killed because of no output) https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received

**Comment by dcoutts** [comment]
I don't think I like the name.

I appreciate that it's short and vaguely funny, but it's a pun on CBOR and CBOR is not a sufficiently well recognised thing yet for that to be a helpful name to help people realise or remember what it is or does.

`binary-serialise` at least says what it does. @hvr complains that it's overly generic. Still I think it's the best one I've seen. Dropping the `-cbor` does make it more generic, but I don't think that's terrible and as I said I don't think cbor is a big marketing pull here. Anyone searching for "cbor" on hackage will still find it from the package synopsis & description.

**Comment by axman6** [comment]
With a little marketing (announcing on -libraries and r/Haskell and some packages picking it up), I don't think it will take people long to catch on given its benefits. IMO `binary-serialise` implies a much more generic library than it actually offers - it doesn't serve the purpose of replacing the binary package at all because it isn't intended for generic binary (de)serialisation. I feel cyborg continues a nice tradition of mild pun names in the Haskell ecosystem which are well known (parsec, and the economy of inspired packages in attoparsec and megaparsec, cassava, thyme, perhaps cabal even falls into this category, etc.) I actually think the pun will make people more likely to remember the name, and encourages people to get an understanding of the format that's being used. My 2¢

**Comment by dcoutts** [comment]
> isn't intended for generic binary (de)serialisation.

Oh, I thought that was exactly what it was for! Do we understand that description in different ways?

**Comment by axman6** [comment]
My point is that I can't arbitrarily encode and decode Protocol X with the library, only CBOR. It's a replacement for binary's haskell data type serialisation, but not binary serialisation in general. As things are, I can't really see how it would be extended for that either, as most of the parsing machinery is quite CBOR specific.

**Comment by axman6** [comment]
Also, I believe the choice of cborg was basically based on your comment https://github.com/well-typed/binary-serialise-cbor/issues/26#issuecomment-222136577 which 7 others also showed their agreement with :)

**Comment by axman6** [comment]
If this is happening, this repo's name will also need to be renamed (GitHub does a great job of handling renames, automatically forwarding URLs, including git URLs).

**Comment by bgamari** [comment]
Here it goes.

**Pull request merged**